open-apparel-registry/open-apparel-registry
# Add extended field reports

## Overview

Adds new reports for extended fields:

- Contributors With Extended Fields Cumulative: This report lists
the number of contributors who have submitted each field type. It is
organized into months, where each month's count includes all extended
fields with an update date of that month or earlier.
- Facilities With Extended Fields Cumulative: This report lists
the number of facilities that have each field type. It is
organized into months, where each month's count includes all extended
fields with an update date of that month or earlier.
- Processing Type Facility Type Matched Values: This report lists each
taxonomy value that has been assigned to an extended field, accompanied
by a list of the raw values that have been matched to that value (separated
by '|').
- Processing Type Facility Type Unmatched Values: This report lists
each raw value that was submitted as a Processing_Type_Facility_Type
and wasn't matched to a taxonomy value, as well as a count of the number
of times that raw value was submitted.
- Submitted Product Type Values: This report lists each raw value that
was submitted for a product_type extended field, as well as a count of
the number of times that raw value was submitted.

Connects #1767 

## Demo

<img width="1100" alt="Screen Shot 2022-04-29 at 2 43 15 PM" src="https://user-images.githubusercontent.com/21046714/166006960-ad98c044-0077-4889-b8cf-7c31f04cc1a9.png">
<img width="1101" alt="Screen Shot 2022-04-29 at 2 43 24 PM" src="https://user-images.githubusercontent.com/21046714/166006967-80ca62c6-e650-4ecc-bf33-00d9d89261dc.png">
<img width="1192" alt="Screen Shot 2022-04-29 at 2 43 35 PM" src="https://user-images.githubusercontent.com/21046714/166006978-5bb596fa-2cc7-4b2b-9593-3d795a54e80d.png">
<img width="654" alt="Screen Shot 2022-04-29 at 2 43 45 PM" src="https://user-images.githubusercontent.com/21046714/166007005-06daedae-87e6-4c6c-a2c0-9a03611f6148.png">
<img width="463" alt="Screen Shot 2022-04-29 at 2 43 58 PM" src="https://user-images.githubusercontent.com/21046714/166007026-15387461-ffbe-40f5-8dbd-532ebc10a38b.png">

## Notes

The issue requested one report with "Report that lists all taxonomy values for product_type, processing_type_facility_type, and the raw values that were fuzzy matched to these values",
but `product_type` doesn't use any fuzzy matching. It could be that this was meant to be `processing_type`
(which is already included under the `processing_type_facility_type` report), but because the
OAR team has also mentioned wanting to evaluate the `product_type` values being submitted,
I included the `product_type` values as a separate report.

## Testing Instructions

* Run `./scripts/server`
* Visit [contributors_with_extended_fields_cumulative](http://localhost:8081/admin/reports/contributors_with_extended_fields_cumulative)
* Visit [facilities_with_extended_fields_cumulative](http://localhost:8081/admin/reports/facilities_with_extended_fields_cumulative)
* Visit [processing_type_facility_type_matched_values](http://localhost:8081/admin/reports/processing_type_facility_type_matched_values)
* Visit [processing_type_facility_type_unmatched_values](http://localhost:8081/admin/reports/processing_type_facility_type_unmatched_values)
* Visit [submitted_product_type_values](http://localhost:8081/admin/reports/submitted_product_type_values)
* Confirm each report contains the expected values and that the format makes sense

## Checklist
